### DISCUSSION TITLE
Add-PnPTeamsChannel does not create folder in SharePoint

### DISCUSSION THREAD

**Discussion by Leif-Frederiksen** [pnp/powershell#1515] [Q&A]
I have an issue where I use Add-PnPTeamsChannel to create a number of channelse on a Team.
Problem is, that only some of the channels are reflected by folders on the corresponding SharePoint site, and the rest are created as channels in Teams, but have no corresponding folder in SharePoint.

Anyone else having this experience and/or suggestions to resolve this?

**Comment by cosibox** [comment]
same experience!

**Comment by maeni70** [comment]
And same experience for me as well! Generally, when running Add-PnPTeamsChannel corresponding folder in SharePoint is not created, and getting "There was a problem reaching this app" when trying to show documents in Files tab of corresponding channel.

Additionally, the same issue occurs in General channel for new teams created by Add-PnPTeamsTeam.

Might this problem be linked to missing permissions? Or is it a bug? No errors are returned when running the cmdlets.

**Comment by gfessl** [comment]
Same for me as well. Unattended app-only authentication (PnP.PowerShell v 1.11) => after running Add-PnPTeamsChannel with private channel configuration, the Private Channels SiteCollection is missing. No error message was shown.

**Reply by gfessl** [reply]
Principally this should be possible according to the documentation (application permission: Channel.Create): https://learn.microsoft.com/en-us/graph/api/channel-post?view=graph-rest-1.0&tabs=http.

This topic should be related: https://github.com/microsoftgraph/microsoft-graph-docs/issues/10206

But the mentioned workaround - polling the filesFolder Endpoint - isn't successfull after more than 3 days...

**Reply by gfessl** [reply]
Finally got it working following the approach mentioned in the link. Polling the filesFolder using invoke-restmethod or invoke-pnpgraphmethod after the Add-PnPTeamsChannel command did the trick.

**Reply by glasserre-pdw** [reply]
Hello @gfessl 
Can you details what you do when you say : 
"Polling the filesFolder using invoke-restmethod or invoke-pnpgraphmethod after the Add-PnPTeamsChannel command"
Thanks in advance.
